client/src/pages/Challenge.tsx

```diff
@@ -0,0 +1,145 @@
+        "expectedFunctionality": {
+          "ModelCreation": "Design comprehensive urban models with diverse components including residential areas, commercial zones, transportation networks, and public services.",
+          "ImpactAnalysis": "Analyze the effects of proposed developments on city infrastructure, such as traffic congestion, environmental sustainability, and public service accessibility.",
+          "SimulationVariants": "Generate and compare different development scenarios to identify optimal urban layouts and policies."
+        },
+        "usageScenarios": {
+          "CloningForExpansionProjects": "Urban planners use UrbanSim to clone the existing urban model as a baseline for proposed expansions. This cloned model is then augmented with new residential areas, commercial centers, and transportation networks. This enables a direct comparison of potential future developments against the current state.",
+          "NewPublicTransportSystem": "Planners simulate the introduction of a new public transport system, evaluating its effects on traffic flow and commuter times. The simulation includes various transport modes and their integration into the existing urban fabric.",
+          "UrbanRenewalProjects": "UrbanSim models the impact of a major urban renewal project, providing insights into potential challenges and benefits for the community. This includes revitalizing old districts, adding green spaces, and improving public amenities.",
+          "PopulationGrowthAdaptation": "Planners use UrbanSim to anticipate changes due to population growth, simulating the expansion of housing, schools, hospitals, and transportation systems to maintain quality of life for residents."
+        }
+      };
+
+    if (details == undefined){
+        setDetails(challengeDetails);  
+    } 
+    
+    //END FOR TESTING ONLY
+
+    if (details != undefined)
+    return(
+    <Container>
+        <section>
+        <Row>
+            <header className="text-center  bg-dark text-light p-5 my-4">
+                <h1 className="fw-semibold" style={{color: getDifficulty(details.difficulty)}} >{details.title}</h1>
+                <h2 className="text-light fw-normal fs-3">{details.outcome}</h2>
+            </header> 
+        </Row>
+        <Row>
+            {/* If completed, allow to view design patterns. */}
+            {completed && <div className="w-100 mb-4 text-start">
+                    {/* <p className="mx-5 text-success"><strong>Completed</strong></p> */}
+                    {/* toggle the state on click and hide/display design pattern list*/}
+                    {!showingDetails ? 
```

Comment:
This part might look better as an [accordion](https://getbootstrap.com/docs/5.3/customize/color/)